src/Datadog.Trace/Tracer.cs

```diff
@@ -212,15 +212,20 @@ public static Tracer Instance
         {
             get
             {
-                return LazyInitializer.EnsureInitialized(ref _instance, ref _globalInstanceInitialized, ref _globalInstanceLock);
+                _globalManualReset.Wait();
+
+                lock (_globalInstanceLock)
```

Comment:
locks should not be needed if you we ResetEvents for synchronization

Reply:
These were there for the 2x set synchronization (`Initialize `and property `set`). ResetEvent should be for get & set synchronization. Theoretically only tests would be affected.